### PR TITLE
add new package : busybox@1.31.1

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/busybox/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Busybox(Package):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/busybox/package.py]:
```suggestion
class Busybox(MakefilePackage):
```

File: var/spack/repos/builtin/packages/busybox/package.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    the utilities you usually find in GNU fileutils, shellutils, etc"""
+
+    homepage = "https://busybox.net"
+    url      = "https://busybox.net/downloads/busybox-1.31.0.tar.bz2"
+
+    version('1.31.1', sha256='d0f940a72f648943c1f2211e0e3117387c31d765137d92bd8284a3fb9752a998')
+
+    def install(self, spec, prefix):
+        make('defconfig')
+        make()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/busybox/package.py]: Put these first two in a `build()` function, the last two in `install()`

File: var/spack/repos/builtin/packages/busybox/package.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        make('install')
+        install_tree('', prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/busybox/package.py]:
```suggestion
        install_tree('.', prefix)
```

File: var/spack/repos/builtin/packages/busybox/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/busybox/package.py]: Thanks, I have already refine it in the new commit.

File: var/spack/repos/builtin/packages/busybox/package.py (hunk 2 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/busybox/package.py]: Thanks, I have already refine it in the new commit.

File: var/spack/repos/builtin/packages/busybox/package.py (hunk 3 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/busybox/package.py]: Thanks, I have already refine it in the new commit.

**Pull request merged**